src/relay.js

```diff
@@ -107,6 +107,115 @@ export function nodeType(connectionType) {
+}
+
+function getOrder(orderBy, options) {
+  const {model} = options;
+  const result = orderBy.map(([orderAttr, orderDirection]) => [
+    orderByAttribute(orderAttr, options),
+    orderDirection,
+  ]);
+  model.primaryKeyAttributes.forEach(primaryKeyAttribute => {
+    if (result.findIndex(([attr]) => attr === primaryKeyAttribute) < 0) {
+      result.push([primaryKeyAttribute, 'ASC']);
+    }
+  });
+  return result;
+}
+
+function reverseDirection(direction) {
+  return direction.indexOf('ASC') >= 0
+    ? direction.replace('ASC', 'DESC')
+    : direction.replace('DESC', 'ASC');
+}
+
+function reverseOrder(order) {
+  return order.map(([orderAttr, orderDirection]) => [orderAttr, reverseDirection(orderDirection)]);
+}
+
+
+/**
+ * Creates a cursor given a node returned from the Database
+ * @param  {Object}   node            sequelize model instance
+ * @param  {String[]} orderAttributes  the attributes pertaining in ordering
+ * @return {String}                   The Base64 encoded cursor string
+ */
+function toCursor(node, info) {
+  return base64(JSON.stringify(info.orderAttributes.map(attr => node.get(attr))));
+}
+
+/**
+ * Creates a cursor given a node returned from the Database
+ * (in the case that an offset had to be used instead of a window)
+ * @param  {Object}   node            sequelize model instance
+ * @param  {String[]} orderAttributes  the attributes pertaining in ordering
+ * @return {String}                   The Base64 encoded cursor string
+ */
+function toOffsetCursor(node, index) {
+  const {primaryKeyAttribute} = getModelOfInstance(node);
+  const id = typeof primaryKeyAttribute === 'string' ? node.get(primaryKeyAttribute) : null;
+  return base64(JSON.stringify([id, index]));
+}
+
+/**
+ * Decode a cursor into its component parts
+ * @param  {String} cursor Base64 encoded cursor
+ * @return {any[]}         array containing values of attributes pertaining to ordering
+ */
+function fromCursor(cursor) {
+  return JSON.parse(unbase64(cursor));
+}
+
+function getWindow({model, cursor, order, inclusive}) {
+  const values = fromCursor(cursor);
+  order.forEach(([orderAttribute], index) => {
+    if (model.rawAttributes[orderAttribute].type instanceof model.sequelize.constructor.DATE) {
+      values[index] = new Date(values[index]);
+    }
+  });
+  const $or = [];
+  const window = {$or};
+
+  if (inclusive) {
+    // include any rows with all values equal to the cursor
+    const isEqualToCursor = {};
+    values.forEach((value, i) => {
+      const [attr] = order[i];
+      isEqualToCursor[attr] = value;
+    });
+    $or.push(isEqualToCursor);
+  }
+
+  // include any rows that are beyond the cursor
+
+  // given ORDER BY A ASC, B DESC, C ASC, the following code would create this logic:
+  // (A > cursorValues[A]) OR
+  // (A = cursorValues[A] AND B < cursorValues[B]) OR
+  // (A = cursorValues[A] AND B = cursorValues[B] AND C > cursorValues[C])
```

Comment:
@janmeier I don't suppose there's a better or more idiomatic way to do this, would you know?  I haven't actually checked how well an index on (A, B, C) would support this, though I have to think it would...

Reply:
@janmeier I saw this syntax for MySQL at least in an article:

```sql
select
  t.id, 
  t.value,
  case (
      select count(*)
      from t t2
      where (t2.value, t2.id) <= (t.value, t.id)
    ) % 5 
    when 0 then 1 
    else 0 
  end page_boundary
from t
order by t.value, t.id
```